app/src/behaviors/behavior_hold_tap.c

```diff
@@ -632,7 +656,10 @@ static int position_state_changed_listener(const zmk_event_t *eh) {
 
     LOG_DBG("%d capturing %d %s event", undecided_hold_tap->position, ev->position,
             ev->state ? "down" : "up");
-    capture_event(eh);
+    struct captured_event capture;
+    capture.tag = ET_POS_CHANGED;
+    memcpy(&capture.data.position, eh, sizeof(struct zmk_position_state_changed_event));
```

Comment:
It seems weird to have used `ev = as_zmk_position_state_changed(eh)` earlier, but then have to go back to using `eh` in order to copy the whole event. I feel like that could be error-prone, since there's no obvious way to copy an event in a type-safe way.

Maybe we could generate a function per event that uses CONTAINER_OF to get a `struct <event_type>_event*` from a `struct <event_type>*`, so then you could use assignment instead of memcpy? Then the compiler would be able to ensure that the types match.

```suggestion
    struct captured_event capture = {
      .tag = ET_POS_CHANGED,
      .data = { .position = *zmk_position_state_changed_container(ev) },
    };
```

(Not really sure what to call said function.) The main downside to this is that there's no guarantee that a `struct <event_type>` is actually contained within a `struct <event_type>_event`. 

Alteratively, if we only ever need to use that to copy a whole event (which is the case as far as I've gotten on this review so far), it could be called "copy" and return the whole event by value instead of as a pointer to the event's container. This has the same problem with if the data isn't part of an event, but it makes the purpose of the function clearer.

Alternatively (and this would require changing a *lot* of other code, but it might we worth it for making copying events simpler and safer), change the `as_<event_type>()` functions to return a pointer to the `_event` struct instead of just the data, and then no extra function is needed to copy the event, but you do have to write a bunch of `.data` everywhere.

Reply:
Ok, I took your idea of a the `CONTAINER_OF` approach and am now generating `copy_raised_foo` functions. I included the `raised_` portion to try to somewhat address the "won't work if not actually contained in a <event_type>_event" issue at least in the naming to make it clear.

Updated the few spots (hold taps and combos) to remove the use of `memcpy` and instead call the generated functions as you suggested. 

Looks a *lot* cleaner to me, but would love your take on the updated version. Thanks!